extra/1-radio-stations.js

```diff
@@ -1,97 +1,116 @@
- *
- * This function should:
- * - Get the available frequencies from `getAllFrequencies`
- * - There is a helper function called isRadioFrequency that takes an integer as an argument and returns a boolean.
- * - Return only the frequencies that are radio stations.
- */
-// `getStations` goes here
-
-/* ======= TESTS - DO NOT MODIFY ======= */
-
-function getAvailableStations() {
-  // Using `stations` as a property as defining it as a global variable wouldn't
-  // always make it initialized before the function is called
-  if (!getAvailableStations.stations) {
-    const stationCount = 4;
-    getAvailableStations.stations = new Array(stationCount)
-      .fill(undefined)
-      .map(function () {
-        return Math.floor(Math.random() * (108 - 87 + 1) + 87);
-      })
-      .sort(function (frequencyA, frequencyB) {
-        return frequencyA - frequencyB;
-      });
-  }
-
-  return getAvailableStations.stations;
-}
-
-function isRadioStation(frequency) {
-  return getAvailableStations().includes(frequency);
-}
-
-const assert = require("assert");
-
-function test(testName, fn) {
-  try {
-    fn();
-    console.log(`\n✅ ${testName}: PASS`);
-  } catch (error) {
-    console.log(
-      `\n❌ ${testName}: FAIL (see details below)\n\n${error.message}`
-    );
-  }
-}
-
-test("getAllFrequencies() returns all frequencies between 87 and 108", function () {
-  const frequencies = getAllFrequencies();
-  assert.deepStrictEqual(frequencies, [
-    87,
-    88,
-    89,
-    90,
-    91,
-    92,
-    93,
-    94,
-    95,
-    96,
-    97,
-    98,
-    99,
-    100,
-    101,
-    102,
-    103,
-    104,
-    105,
-    106,
-    107,
-    108,
-  ]);
-});
-
-test("getStations() returns all the available stations", () => {
-  const stations = getStations();
-  assert.deepStrictEqual(stations, getAvailableStations());
-});
+/**
+ * Finding a radio station, and a good one, can be hard manually.
+ * Let's use some code to help us build a program that helps us scan
+ * the radio waves for some good music.
+ */
+
+/**
+ * First, let's create a function that creates a list of all the frequencies.
+ * Call this function `getAllFrequencies`.
+ *
+ * This function should:
+ * - Create an array  starting at 87 and ending in 108
+ * - Should return this array to use in other functions
+ */
+
+// `getAllFrequencies` goes here
+
+function getAllFrequencies() {
+  let radioFrequencies = [];
+  for (let i = 87; i <= 108; ++i) {
+    radioFrequencies.push(i);
+  }
+  return radioFrequencies;
+}
+
+/**
+ * Next, let's write a function that gives us only the frequencies that are radio stations.
+ * Call this function `getStations`.
+ *
+ * This function should:
+ * - Get the available frequencies from `getAllFrequencies`
+ * - There is a helper function called isRadioFrequency that takes an integer as an argument and returns a boolean.
+ * - Return only the frequencies that are radio stations.
+ */
+// `getStations` goes here
+
+
+function getStations() {

```

Comment:
So on this question I doubt this is what they wanted hence I high-key believe this may not be passing all test cases...

### Here's an idea behind this function

1. You have created a function **getAllFrequencies** which returns an array back to you
2. Now what you need from this function called **getStations** you need to get **available frequencies from getAllFrequencies function that you created**
3. Now you can call the **getAllFrequencies** function save the returned array in a variable resulting to you having an array of all frequencies which you returned in that function **getAllFrequencies** 
4. Now they are telling us in the instructions that they made a helper function called **isRadioFrequency** now this function will tell **true** if the frequency is a radio station or **false** if not 
5. Now you have interact with that helper function to help you clean up the array containing all **frequencies** to only remain with frequencies that are for a radio station
6. Finally return all **radio stations**

## Hope this helps 🔥💪

Reply:
Hi Ntsembo

Thank you for the info and help, I did not finish this but will try to finish this with your advise if time allow it.

Reply:
@Jontas . Sorry for that. It's just a difference in answers and I have realised that mine is wrong since many people are going for yours.